setup.py

```diff
@@ -518,7 +519,7 @@ def setup_package():
         classifiers=[_f for _f in CLASSIFIERS.split('\n') if _f],
         platforms=["Windows", "Linux", "Solaris", "Mac OS-X", "Unix"],
         test_suite='pytest',
-        version=versioneer.get_version(),
+        version=VERSION,
```

Comment:
This is a change, because `VERSION` is defined as
```
VERSION = '{}.{}.{}'.format(MAJOR, MINOR, MICRO)
```
and `versioneer.get_version()` returns the version including git hash. And it results in:
```
$ python setup.py sdist
numpy-2.0.2.0.0.dev0+git20230811.8287648.tar.gz
```
easy enough to fix it looks like, by using `FULLVERSION` instead.

Reply:
Ugh, that didn't do it:
```
ERROR: Could not find a version that satisfies the requirement numpy (from versions: 2.0.0.dev0-git20230811.6c254b42.0.0.dev0-git20230811.6c254b4)
```
I'll do some more testing locally. I really want to merge this PR, but this seems like a blocker. Shouldn't be too difficult to fix.

Reply:
Hmm, locally I get sdist and wheel that are named sensibly:
```
numpy-2.0.0.dev0+807.gf015fe226-cp39-cp39-linux_x86_64.whl
```

Reply:
Okay, whatever, CI was green so I'm reverting to what it was. I really do get `numpy-2.0.2.0.0.dev0+git20230811.6aff0bc` locally, no idea what is going on there. But we're switching all jobs away from `setup.py` asap, so maybe it doesn't matter.

Reply:
I've been building wheels using `python -m build --sdist`. Is it necessary to support another pathway?

Reply:
(I am unable to build sdists the other way because of https://github.com/numpy/numpy/issues/24401)

Reply:
That will use Meson, not `setup.py`. It's longer term not necessary to have more than one way, but this change does break something for `setup.py` users I think. Which we can say is fine - time to move over to Meson if this bothers you:)

Reply:
Ah, I think that's a known issue indeed

Reply:
Another line in `setup.py` that doesn't seem quite right:

```
def get_docs_url():
    if 'dev' in VERSION:
        return "https://numpy.org/devdocs"
```

Another reason I think that your FULLVERSION patch is correct.

Reply:
Ah well, let's leave that for a follow-up. I'm kinda running out of steam and have some non open source plans for the weekend.

Reply:
The `use_wheel` job is happy to install this wheel, I guess it's a valid version number:
```
+ pip install --pre --no-index --upgrade --find-links=. numpy
Looking in links: .
Processing ./numpy-2.0.2.0.0.dev0+git20230811.f30bdab-cp39-cp39-linux_x86_64.whl
Installing collected packages: numpy
```

Reply:
2.0.2.0.0 🤪